src/main/kotlin/no/nav/syfo/melding/api/MeldingApi.kt

```diff
@@ -104,6 +106,39 @@ fun Route.registerMeldingApi(
 
             call.respond(HttpStatusCode.OK)
         }
+
+        post("/{$uuid}/retur") {
+            val personIdent = call.personIdent()
+            val veilederIdent = call.getNAVIdent()
+            call.checkVeilederTilgang(
+                action = API_ACTION,
+                veilederTilgangskontrollClient = veilederTilgangskontrollClient,
+                personIdent = personIdent
+            )
+
+            val meldingUuid = call.meldingUuid()
+            val innkommendeLegeerklaring = meldingService.getMeldingFraBehandler(meldingUuid)
+                ?.takeIf { it.type == MeldingType.FORESPORSEL_PASIENT_LEGEERKLARING }
+                ?: throw IllegalArgumentException("Failed to create retur av legeerklæring: Melding with uuid $meldingUuid does not exist")
+            val opprinneligForesporselLegeerklaring = meldingService.getUtgaendeMeldingerInConversation(
+                conversationRef = innkommendeLegeerklaring.conversationRef,
+                personIdent = innkommendeLegeerklaring.arbeidstakerPersonIdent,
+            ).first { it.type == MeldingType.FORESPORSEL_PASIENT_LEGEERKLARING }
+
+            val requestDTO = call.receive<ReturAvLegeerklaringRequestDTO>()
+            val returAvLegeerklaring = requestDTO.toMeldingTilBehandler(
+                opprinneligForesporselLegeerklaring = opprinneligForesporselLegeerklaring,
+                innkommendeLegeerklaring = innkommendeLegeerklaring,
+                veilederIdent = veilederIdent,
+            )
+
+            meldingService.createReturAvLegeerklaring(
+                callId = getCallId(),
+                returAvLegeerklaring = returAvLegeerklaring,
+            )
```

Comment:
Slik jeg tenker så er det servicen sitt ansvar å anvende DTO objekter og eventuelt gjøre om til gyldige domeneobjekter. Hva tenker du om å sende DTO'en rett inn i service metoden her? 
```suggestion
            meldingService.createReturAvLegeerklaring(
                callId = getCallId(),
                returAvLegeerklaring = requestDTO,
            )
```

Reply:
Det vil da se slik ut i `MeldingService`
```kotlin
internal suspend fun createReturAvLegeerklaring(
        callId: String,
        returAvLegeerklaringDTO: ReturAvLegeerklaringRequestDTO,
        opprinneligForesporselLegeerklaring: MeldingTilBehandler,
        innkommendeLegeerklaring: MeldingFraBehandler,
        veilederIdent: String,
    ) {
        val returAvLegeerklaring = MeldingTilBehandler.createReturAvLegeerklaring(
            returAvLegeerklaringDTO,
            opprinneligForesporselLegeerklaring,
            innkommendeLegeerklaring,
            veilederIdent,
        )
        val pdf = createPdf(callId = callId, meldingTilBehandler = returAvLegeerklaring)
        createMeldingTilBehandlerAndSendDialogmeldingBestilling(
            meldingTilBehandler = returAvLegeerklaring,
            pdf = pdf
        )
    }
```

Reply:
Hm, jeg ville egentlig tenkt motsatt - at domene-servicen ikke skal være avhengig av api-laget, men at det er api-laget som bruker domene og domene-servicen. Men forsåvidt enig i at DTOen ikke burde vite hvordan man skal konstruere et gyldig domene-objekt. Det kunne man f.eks. løst ved at DTOen bruker en `MeldingTilBehandler.returLegeerklaring(..): MeldingTilBehandler`-metode (med nødvendige parametere) som har ansvar for å lage nytt `MeldingTilBehandler`-objekt (istedet for at DTOen bruker konstruktøren direkte).

Reply:
Japp, tenkte litt på det der i lys av at vi snakket mye om det å plassere logikk i riktige "lag" før ferien 👌🏼 Det var _noe_ som gjorde at jeg ikke flyttet det inn da, men jeg skal se på det igjen! Mulig det var bare at man ville kaste feil, men det kan man jo gjøre i servicen uansett 🤷🏼‍♂️ Så synes det du foreslår ser bra ut 🙌🏼

Reply:
> Hm, jeg ville egentlig tenkt motsatt - at domene-servicen ikke skal være avhengig av api-laget, men at det er api-laget som bruker domene og domene-servicen.

Tenker du på `MeldingService` som domene-service? I så fall lurer jeg på om vi snakker litt forbi hverandre. Jeg tenker på `MeldingService` her som applikasjons/service lag service. Tenker litt i [onion architecture](https://marcoatschaefer.medium.com/onion-architecture-explained-building-maintainable-software-54996ff8e464) baner.
Vet dere hvilken arkitektur vi har hatt i tankene når vi har satt opp filstrukturen her? 😊

Reply:
Ja, jeg og Anders snakket litt om artikkelen du sendte nå. Vi har nok mest av application-services, men tenkt at det og domain-services er ish det samme. Så tror ikke vi har så mange domain-services rundt omkring egentlig 🤔 Jeg har bare tenkt på services som businesslogikk ved bruk av domeneobjekter, har ikke tenkt på at man kan/burde skille mellom to ulike services. Vi gjør jo som regel database-kall i servicene våre, som man ikke ville gjort i en domain-service.